docs/reference/language/context.md

```diff
@@ -25,38 +25,180 @@ in some places that are also aware of their location in the document:
+results because we created two different contexts.
+
+However, immutability only applies to the property fields themselves. 
+Content creation instructions within a context _do_ see the effect of 
+the set rule. Consider the same example with font size:
+
+```example
+#set text(size: 50pt)
+#context [
+  call 1: #text.size \
+
+  #set text(size: 25pt)
+  call 2: #text.size
+]
+```
+
+Call 2 still outputs '50pt', because `text.size` is a constant.
+However, this output is printed in '25pt' font, as specified by the set
+rule before the call. This illustrates the importance of picking the 
+right insertion point for a context to get access to precisely the right 
+styles. If you need access to updated property fields after a set rule, 
+you can use nested contexts:
+
+```example
+#set text(lang: "en")
+#context [
+  call 1: #text.lang \
+
+  #set text(lang: "fr")
+  call 2: #context text.lang
+]
+```
+
+All of the above applies to `show` rules analogously, for example:
+
+```example
+#let template(body) = {
+  set text(size: 25pt)
+  body
+}
+
+#set text(size: 50pt)
+#context [
+  call 1: #text.size \
+
+  #show: template
+  call 2: #text.size \
+  call 3: #context text.size
+]
+```
+
+## Controlling content creation within a context
+
+The main purpose of retrieving the current values of properties is, 
+of course, to use them in the calculation of derived properties, 
+instead of setting those properties manually. For example, you can
+double the font size like this:
+
+```example
+#context {
+  // the context allows you to
+  // retrieve the current text.size
+  set text(size: text.size * 200%)
+  [large text \ ] 
+}
+original size
+```
 
-#set text(lang: "fr")
-#value
+Since set rules are only active until the end of the enclosing scope, 
+'original size' is printed with the original font size.
+The above example is equivalent to 
+
+```example
+#{
+  set text(size: 2em)
+  [large text \ ]
+}
+original size
 ```
 
-Crucially, upon creation, `value` becomes opaque [content] that we cannot peek
-into. It can only be resolved when placed somewhere because only then the
-context is known. The body of a context expression may be evaluated zero, one,
-or multiple times, depending on how many different places it is put into.
+but convenient alternatives like this do not exist for most properties.
+This makes contexts a powerful and versatile concept. For example, 
+to double the spacing between the lines of an equation block, you can 
+use the same resizing technique in a show rule. In this case, explicitly
+adding the `context` keyword is not necessary, because a show rule
+establishes a context automatically:
+
+```example
+#let spaced-eq(spacing: 100%, body) = {
+  show math.equation.where(block: true): it => {
+    // access current par.leading in the
+    // context of the show rule
+    set par(leading: par.leading * spacing)
+    it
+  }
+  body
+}
+
+normal spacing:
+$
+x \
+x
+$
+doubled spacing:
+#spaced-eq(spacing: 200%)[$
+z \
+z
+$]
```

Comment:
This example is a bit to complex for my liking. I think the point can be made in a clearer way.

Reply:
Would it be better to write `$ x \ x $` ?